### PR TITLE
[style] update case listing card

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
- Updates case listing card according to the new hifi designs

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- fixes scrolling for the cases page
- adds `Icon.tsx` file under an `assets` folder 
- Updates tag color for tags on case listing card
- Updates page title from `h1` to `h2` (change not visible in gif below)
- Updates `getAllCases` and `getNCases` queries to use a database function
- Fixes `Button.tsx` to get rid of console.log warnings

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

caseListingUpdate

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Fetch this branch and `git checkout alvaro/ijp-57-update-case-listing-card`
- Go to `http://localhost:3000/cases` and make sure that the new styling for the case listing components is consistent with the hifi designs 
- Make sure that scrolling doesn't look wonky (@davidqing6432 let me know your thoughts on the scrollbars, @jinkang-0 mentioned potentially styling them so I'm curious what your thoughts on that are)

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Wait on legalserver info regarding format of languages before deciding on a format & conversions for our project, then do any mapping/conversions if necessary

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
- Helpful resource for creating the cases page layout using grids: https://medium.com/@beyondborders/beginner-css-grid-sticky-navigation-scrolling-content-7c4de0a8d1dc

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/styles/text.tsx
Comment (varortz) [inline src/styles/text.tsx]: Same thing here as my other PR, needed global styles for text so I just went ahead and copied over what Catherine++ added in her PR.

**Comment by jinkang-0** [comment]
I think we want the card column to be scrollable as opposed to the whole page with the new design, so probably give `CardColumn` an `overflow` property?

Though this is probably another sprint

**Comment by jinkang-0** [comment]
One thought: I think the current design is to have a scrollbar next to the big card for scrolling it. With the current frontend design, this will limit the scrollbar to be shorter (confined to the `MainDisplay` component).

Would it be cleaner to have the scrollbar for the body (entire page) (i.e. rn) sync to the big card display instead? That won't involve anything from either `body` or `CardDisplay`, since it will naturally expand. We'll just have to make sure the card column is fixed to the screen.

Let's discuss this tomorrow with @davidqing6432 !
(probably also another sprint, maybe the same one)

**Comment by varortz** [comment]
> I think we want the card column to be scrollable as opposed to the whole page with the new design, so probably give `CardColumn` an `overflow` property?

I was planning on addressing this in my next sprint, but tbh it does make more sense to add that fix in this PR so I'll go ahead and include it!

**Comment by jinkang-0** [comment]
I think we want the center divider to be centered between the two and use the mid grey color if possible!

image
image

File: src/app/cases/styles.tsx
Comment (jinkang-0) [inline src/app/cases/styles.tsx:10]: Design Review (placed here to allow for threading)
---

I think we want the center divider to be centered between the two and use the mid grey color if possible!

Current | Figma
:--------:|:-------:
image | image

Comment (jinkang-0) [inline src/app/cases/styles.tsx:10]: I think we should also shift from the current box shadow on the big card to the bordered design!

Current | Figma
:--------:|:-------:
image | image

Comment (varortz) [reply src/app/cases/styles.tsx:10]: this should be fixed in my other PR #25!

File: src/components/ListingCard/styles.tsx
Comment (jinkang-0) [inline src/components/ListingCard/styles.tsx]: This is my fault for not using `rem` before, but do you mind changing the `px` units to `rem` here? 😅

File: src/components/ListingCard/index.tsx
Comment (jinkang-0) [inline src/components/ListingCard/index.tsx]: If a case doesn't need an interpreter or an attorney, do we want to show it? (also might want to remove the `!` before the `needs_attorney` boolean here)

Comment (jinkang-0) [reply src/components/ListingCard/index.tsx]: Also unsure if we'll ever encounter a case where they will find an interpreter before an attorney, but good to have just in case 👍

File: src/styles/colors.ts
Comment (jinkang-0) [inline src/styles/colors.ts]: I think on Figma, David calls it `light primary`, do we want to stick with that or call it `primaryLight`?

File: src/components/ListingCard/styles.tsx
Comment (jinkang-0) [inline src/components/ListingCard/styles.tsx]: Let's go a little crazier, say `3px` (`0.1875rem`)

File: src/components/ListingCard/index.tsx
Comment (jinkang-0) [reply src/components/ListingCard/index.tsx]: Also to consider: could we make a function that helps us with the logic? It could help declutter it since it is technically just one tag with the text conditional on the boolean. (see [PR28](https://github.com/calblueprint/immigration-justice-project/pull/28/files#diff-2b850697f022cdb66b964e47c0f8eee6b3f317f0d63f8d4269c895a5d64cf306), `ButtonDropdown.tsx`)

**Review by jinkang-0 — COMMENTED** [review]
Looks great so far!! I added some minor tweaks and things to consider, but should be good to go after iterations!

File: src/components/ListingCard/index.tsx
Comment (pragyakallanagoudar) [reply src/components/ListingCard/index.tsx]: > If a case doesn't need an interpreter or an attorney, do we want to show it? (also might want to remove the ! before the needs_attorney boolean here)

yeah let's not show it if it doesn't need either!

File: src/styles/colors.ts
Comment (pragyakallanagoudar) [reply src/styles/colors.ts]: is this not called `blueLight` in #30

File: src/components/ListingCard/index.tsx
Comment (pragyakallanagoudar) [reply src/components/ListingCard/index.tsx]: > Also unsure if we'll ever encounter a case where they will find an interpreter before an attorney, but good to have just in case 👍

definitely possible! if, say, more interpreters indicate interest and we handle their interest while waiting for attorney interest or something

File: src/components/ListingCard/styles.tsx
Comment (pragyakallanagoudar) [inline src/components/ListingCard/styles.tsx:8]: question: do we want the listing card to take on different heights depending on how big the tags are? seems like that's what we're doing rn, but it potentially feels cleaner to always have them be a certain height (cc @davidqing6432)

File: src/components/ListingCard/index.tsx
Comment (pragyakallanagoudar) [inline src/components/ListingCard/index.tsx]: we want a null check here! something like
`caseData.upcoming_date ? parseDate(timestampStringToDate(caseData.upcoming_date)) : 'N/A'`?

File: src/app/cases/styles.tsx
Comment (pragyakallanagoudar) [inline src/app/cases/styles.tsx:19]: would it be possible for us to change things to just have the column of case listing cards be scrollable? especially now that the interest form seems like it will generally be scrollable....i think having both of them being independently scrollable is the move, as opposed to what it currently is which is the whole page being scrollable and having that scroll through the listing cards?

@varortz i know you mentioned changing this in an interest form PR but lowk it might be ideal to add it in here?

(cc @jinkang-0 @davidqing6432)

**Review by pragyakallanagoudar — COMMENTED** [review]
LOOKS SO GOOD ALVAROOOO TY FOR WORKING ON THIS

next steps on this: 
- once #29 is merged in, this will likely have quite a few changes :') 
- once david finalizes the listing card (i.e. tags, location icon, etc.), some design changes

File: src/styles/colors.ts
Comment (jinkang-0) [reply src/styles/colors.ts]: yes this is outdated comment, i shall resolve

File: src/app/cases/styles.tsx
Comment (jinkang-0) [reply src/app/cases/styles.tsx:19]: yes agreed, we may want an updated design to account for the added width of the scrollbars though @davidqing6432 (perhaps even a styled scrollbar given how ugly it is on windows)

File: src/components/ListingCard/index.tsx
Comment (varortz) [reply src/components/ListingCard/index.tsx]: sorry, that `!` was a typo

File: src/components/ListingCard/styles.tsx
Comment (varortz) [reply src/components/ListingCard/styles.tsx]: Wondering if there's a reason you think we should do `3px` instead? I think David wants `2px` borders for most components, and this is also currently consistent with the Figma design

File: src/components/ListingCard/styles.tsx
Comment (varortz) [reply src/components/ListingCard/styles.tsx:8]: Solved for now? Talked to @davidqing6432 during worksesh today, and decided on increasing the width of the listing cards to make the size difference less obvious. Let me know if we wanna go for a different approach though!!

File: src/components/ListingCard/index.tsx
Comment (varortz) [reply src/components/ListingCard/index.tsx]: solved!

File: src/app/cases/styles.tsx
Comment (varortz) [reply src/app/cases/styles.tsx:10]: updated spacing + color 👍

File: src/components/ListingCard/index.tsx
Comment (varortz) [reply src/components/ListingCard/index.tsx]: Also, glad you mentioned the helper function, I thought of doing that initially and ended up not doing it lol. I've included it now though!

File: src/components/ListingCard/styles.tsx
Comment (jinkang-0) [reply src/components/ListingCard/styles.tsx]: Oh right, I think it was showing up too thin compared to the Figma. `2px` seems to be what it is. Although on Figma it seems to be "tearful blue"?

Comment (varortz) [reply src/components/ListingCard/styles.tsx]: Oh yeah, it's only "tearful blue" when it's selected, this should be on line 19. Otherwise, the color is `greyLight`

File: src/app/cases/styles.tsx
Comment (varortz) [reply src/app/cases/styles.tsx:19]: fixed the scrolling 🥰

**Comment by david-qing** [comment]
lgtm

**Review by david-qing — APPROVED** [review]
great stuff

File: src/app/cases/styles.tsx
Comment (david-qing) [inline src/app/cases/styles.tsx]: this padding-right gets overridden by the following padding (i think), so you can remove this line

File: src/components/ListingCard/index.tsx
Comment (jinkang-0) [inline src/components/ListingCard/index.tsx]: A couple notes, but in short, this code is dangerous and should be removed ASAP.

This code is dangerous because a dependency array is not passed as an argument after the function (callback) to `useEffect`). When a dependency array is absent, React will [run the callback on **every** render](https://legacy.reactjs.org/docs/hooks-effect.html). This, in conjunction with expensive query calls, spell large money costs, as roughly 400 queries are made with each render. That's every time you save a file when devving, navigate back to the cases page (from another page), reload, or even change the props passed into each card. All this is to say, in the future, please add a dependency list for expensive calls, even if an empty one (`[]`).

Instead of adding a dependency list to fix it though, I would suggest using a [foreign table query](https://supabase.com/docs/reference/javascript/select?example=query-foreign-tables-through-a-join-table) when we query for cases (`getNCases`). This allows us to have all the data in one place and reduce the complexity in managing different queries.

**Review by jinkang-0 — CHANGES_REQUESTED** [review]

File: src/components/ListingCard/index.tsx
Comment (varortz) [reply src/components/ListingCard/index.tsx]: nice catch that's my bad, I thought I did have the empty dependency array. Will fix asap!

File: src/app/cases/styles.tsx
Comment (varortz) [reply src/app/cases/styles.tsx]: oops, forgot to remove that!

File: src/components/ListingCard/index.tsx
Comment (pragyakallanagoudar) [inline src/components/ListingCard/index.tsx:18]: these helper functions are beautiful

**Review by pragyakallanagoudar — APPROVED** [review]
AMAZINGGGG ALVARO TYSM FOR YOUR WORK ON THIS!! i think this is ready to go :)) 

some next steps:
- "no cases listed" page with no cards -- how this looks is affected by whether or not the user is authenticated, so that piece can be included in an upcoming sprint that checks whether the user is authenticated / onboarded / etc. and changes the case cards & interest form accordingly
- design changes -- we're hitting the pause button on any other design changes until we get LegalServer access, but just something to keep in mind!!

**Review by jinkang-0 — APPROVED** [review]

File: src/components/ListingCard/styles.tsx
Comment (jinkang-0) [inline src/components/ListingCard/styles.tsx]: should we remove these comments?

File: src/components/ListingCard/index.tsx
Comment (jinkang-0) [inline src/components/ListingCard/index.tsx]: oop I added a similar helper in `src/utils/helpers.ts` (see [multiselect PR](https://github.com/calblueprint/immigration-justice-project/pull/28/files#diff-d9b0865f2e8f841410cd4faf3e1e4a1fb7cb0b545f583f6adaad5841d7eaa58d)) 
do you want to make something similar, or do you want me to make the switch later?

File: src/app/cases/styles.tsx
Comment (jinkang-0) [inline src/app/cases/styles.tsx]: can you put `overflow-y: auto` instead? again, windows issue, but if it's `scroll`, the scrollbar will be permanently there, whereas `auto` only adds it if necessary

File: src/app/cases/styles.tsx
Comment (jinkang-0) [inline src/app/cases/styles.tsx]: `width: 100%` pls

**Review by jinkang-0 — CHANGES_REQUESTED** [review]
looks great!! ~~gonna request changes for some styling quirks, but ping me when they're addressed and I'll approve. i can personally guarantee you i will be available before midnight, if not, be worried.~~

actually nvm, i don't think those changes are too important, so feel free to choose if you want to implement them right now and merge when you're ready

**Review by jinkang-0 — APPROVED** [review]

File: src/components/ListingCard/index.tsx
Comment (varortz) [reply src/components/ListingCard/index.tsx]: no worries, I'll go ahead and update. Ty for bringing it up!

**Pull request merged**
